lib/os/CMakeLists.txt

```diff
@@ -24,6 +24,7 @@ zephyr_sources(
   heap-validate.c
   bitarray.c
   multi_heap.c
+  utf8.c
```

Comment:
Needs a Kconfig option to not slow down the build for all that do not use these functions.

Reply:
@SebastianBoe I can absolutely add that :) 

Should have the same for some of the above as well? I figure that stuff like `printk`, `cbprintf`, `hex.c` and likely other suffer from the same issue, don't they?

Reply:
Yes, but it will be a backwards-incompatible change to add this after-the-fact, so it is a bit awkward. And definitely not something I expect you to do in this PR.

Also some things, that are used by core features, will be enabled so often that we might as well always enable them.

But it's difficult to identify these.